### PR TITLE
Static compiling, better string handling, et al.

### PR DESCRIPTION
**Pull request by rockon999**
The culmination of many branches and experiments.

Moves the site to `Gridsome` for pre-compiling and server-side rendering. This greatly improves speed.

Along the way, several components have been converted to TypeScript because Gridsome is more picky about types.

More to follow.

### CODE REVIEW

**Comment by SamChou19815** [comment]
@rockon999 I cannot find the netlify CI config. You need to add `node cli.js` to the build step before `yarn build`, and ensure that you are on Node >= 12. Otherwise things won't build.

**Comment by SamChou19815** [comment]
@rockon999 ready?

File: config.js [hunk 1]
```diff
@@ -0,0 +1,1590 @@
...
+    net: 'empty',
+    tls: 'empty',
+    child_process: 'empty'
+  },
+  output: {
+    path: '/home/evan/Development/DTI/DTI-NOVA/dti-nova/dist',
+    filename: 'js/[name].js',
+    publicPath: '/',
+    chunkFilename: 'js/[name].js'
+  },
```
Comment (SamChou19815) [inline config.js]: @rockon999 🤔

Comment (rockon999) [reply config.js]: `vue-cli-service` uses an internal webpack config now. I dumped it to figure out some plugin issues, and forgot I dumped it.

**Pull request merged**